### PR TITLE
Allow for components that are not maps (e.g., Java objects)

### PR DESCRIPTION
**Pull request by hlship**
This came up when I was attempting to make a HTTP connection manager a shared component; it felt like it should work (by extending Lifecycle onto the interface the manager implements) BUT got exceptions in schematic related to `vary-meta` on the component.

This fixes that case, checking to see if the component that results from invoking the init-fn is an IObj, and only manipulates its meta-data if so.

As implemented here, dependencies for such a component are quietly ignored (not applied to the component).

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(defproject com.walmartlabs/schematic "1.1.0"
+(defproject com.walmartlabs/schematic "1.2.0-SNAPSHOT-1"
```
Comment (sashton) [inline project.clj]: What is this odd version number or?

File: src/com/walmartlabs/schematic.clj [hunk 2]
```diff
@@ -63,15 +64,22 @@
...
-                       init-fn (init-fn)
-                       (map? v) (vary-meta dissoc ::component/dependencies))]
-      (component/using component' ref-map))
-    v))
+          component' (cond-> (dissoc v :sc/create-fn :sc/refs)
+                       init-fn (init-fn))]
+      (if (instance? IObj component')
+        (-> component'
+            ;; Sanity: clear any existing dependencies already present, though
+            ;; it's not clear why they would be there!
```
Comment (sashton) [inline src/com/walmartlabs/schematic.clj]:
```suggestion
            ;; Such dependencies might exist in the metadata if a plain-Component constructor
            ;; function is being re-used as a Schematic component init function. 
            ;; Schematic does not honor any dependency metadata applied by other means. 
```

**Review by sashton — COMMENTED** [review]
Looks good and makes sense. Just one question and one suggestion.

File: project.clj (hunk 1 above)
Comment (hlship) [reply project.clj]: That was just so I could use the updated code in my other project (Clockwork) which is where I stumbled on this issue.  I can back that out and we can do a proper 1.2.0.

**Review by sashton — APPROVED** [review]

**Pull request merged**